cscs-checks/microbenchmarks/spec-accel/spec.py

```diff
@@ -0,0 +1,121 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+from reframe.core.launchers.registry import getlauncher
+
+
+class SpecAccelCheckBase(rfm.RegressionTest):
+    def __init__(self, prg_envs):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.valid_prog_environs = prg_envs
```

Comment:
Since you set this in the subclasses, I suggest not to use `prg_envs` here at all. Just set the `valid_prog_environs` in the subclasses.

Reply:
`valid_prog_environs` needs to be set after `__init__` of the rfm base class, because it gets overwritten, and before the body of my base's `__init__` because it gets referenced there.
I can only do that by passing an argument.